include/nbl/builtin/hlsl/bxdf/transmission/smooth_dielectric.hlsl

```diff
@@ -0,0 +1,332 @@
+        scalar_type NdotV = interaction.isotropic.getNdotV();
+        bool dummy;
+        return __generate_wo_clamps(interaction.isotropic.getV().getDirection(), interaction.getT(), interaction.getB(), interaction.isotropic.getN(), NdotV, 
+            NdotV, u, orientedEta, rcpEta, dummy);
+    }
+
+    sample_type generate(NBL_CONST_REF_ARG(anisotropic_interaction_type) interaction, NBL_REF_ARG(vector<scalar_type, 3>) u)
+    {
+        fresnel::OrientedEtas<scalar_type> orientedEta = fresnel::OrientedEtas<scalar_type>::create(interaction.isotropic.getNdotV(), eta);
+        fresnel::OrientedEtaRcps<scalar_type> rcpEta = fresnel::OrientedEtaRcps<scalar_type>::create(interaction.isotropic.getNdotV(), eta);
+        scalar_type NdotV = interaction.isotropic.getNdotV();
+        bool dummy;
+        return __generate_wo_clamps(interaction.isotropic.getV().getDirection(), interaction.getT(), interaction.getB(), interaction.isotropic.getN(), NdotV, 
+            nbl::hlsl::abs<scalar_type>(NdotV), u, orientedEta, rcpEta, dummy);
+    }
+    sample_type generate(NBL_CONST_REF_ARG(isotropic_interaction_type) interaction, NBL_REF_ARG(vector<scalar_type, 3>) u)
+    {
+        return generate(anisotropic_interaction_type::create(interaction), u);
+    }
+
+    // eval and pdf return 0 because smooth dielectric/conductor BxDFs are dirac delta distributions, model perfectly specular objects that scatter light to only one outgoing direction
+    scalar_type pdf(NBL_CONST_REF_ARG(params_isotropic_t) params)
+    {
+        return 0;
+    }
+    scalar_type pdf(NBL_CONST_REF_ARG(params_anisotropic_t) params)
+    {
+        return 0;
+    }
+
+    quotient_pdf_type quotient_and_pdf(NBL_CONST_REF_ARG(params_isotropic_t) params)
+    {
+        const bool transmitted = ComputeMicrofacetNormal<vector3_type>::isTransmissionPath(params.getNdotVUnclamped(), params.getNdotLUnclamped());
+
+        fresnel::OrientedEtaRcps<scalar_type> rcpOrientedEtas = fresnel::OrientedEtaRcps<scalar_type>::create(params.getNdotV(), eta);
+
+        const scalar_type _pdf = bit_cast<scalar_type, uint32_t>(numeric_limits<scalar_type>::infinity);
+        scalar_type quo = hlsl::mix<scalar_type, bool>(1.0, rcpOrientedEtas.value, transmitted);
+        return quotient_pdf_type::create((spectral_type)(quo), _pdf);
+    }
+    quotient_pdf_type quotient_and_pdf(NBL_CONST_REF_ARG(params_anisotropic_t) params)
+    {
+        const bool transmitted = ComputeMicrofacetNormal<vector3_type>::isTransmissionPath(params.getNdotVUnclamped(), params.getNdotLUnclamped());
+
+        fresnel::OrientedEtaRcps<scalar_type> rcpOrientedEtas = fresnel::OrientedEtaRcps<scalar_type>::create(params.getNdotV(), eta);
+
+        const scalar_type _pdf = bit_cast<scalar_type, uint32_t>(numeric_limits<scalar_type>::infinity);
+        scalar_type quo = hlsl::mix(1.0, rcpOrientedEtas.value, transmitted);
+        return quotient_pdf_type::create((spectral_type)(quo), _pdf);
+    }
```

Comment:
nope, that is not the correct quotient calculation

you transmit with probablity `(1-reflectance)*deltaDistribution(L=V.refract(N))` and reflect with probability `reflectance*deltaDistributiron(L=v.reflect(N))`

This means your quotient_and_pdf is always (1.f,INF)

Why? There's no absorption, the light reflected and refracted always add up to 1 and you've importance sampled it perfectly.